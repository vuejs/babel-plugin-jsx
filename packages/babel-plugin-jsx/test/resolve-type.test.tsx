import { transformAsync } from '@babel/core';
// @ts-expect-error missing types
import typescript from '@babel/plugin-syntax-typescript';
import VueJsx from '../src';

describe('resolve type', () => {
  describe('runtime props', () => {
    test('basic', async () => {
      const result = await transformAsync(
        `
        interface Props { foo?: string }
        const App = defineComponent((props: Props) => <div />)
        `,
        {
          plugins: [[typescript, { isTSX: true }], VueJsx],
        }
      );
      expect(result!.code).toMatchSnapshot();
    });
  });
});
