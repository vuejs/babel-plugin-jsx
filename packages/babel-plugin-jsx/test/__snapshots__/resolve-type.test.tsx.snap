// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`resolve type > runtime props > basic 1`] = `
"import { createVNode as _createVNode } from "vue";
interface Props {
  foo?: string;
}
const App = defineComponent((props: Props) => _createVNode("div", null, null), {
  props: {
    foo: {
      type: String,
      required: false
    }
  },
  name: "App"
});"
`;
